File: src/recursion/pattern/bubbleSort.java
package recursion.pattern;

import java.util.Arrays;

public class bubbleSort {
    static void bsort(int arr[],int row, int col){
        if(row == 0){
            // base case
            return;
        }

        if(col<row){
            // comparing adjacent no. till row 
            // because after row no. is already sorted 
            if(arr[col]>arr[col+1]){
                // if preceding>succeeding
                int temp = arr[col+1];
                arr[col+1] = arr[col];
                arr[col]=temp;
            }
            // going forward 
            bsort(arr, row, col+1);
        }else{
            // move to next iteration 
            bsort(arr, row-1, 0);
        }
    }
    public static void main(String[] args) {
        int arr[] = {4,3,2,1};
        bsort(arr , arr.length-1, 0);
        System.out.println(Arrays.toString(arr));
    }
}

File: javascript/Sorting/bubbleSort.js
function bSort(array){
    let n = array.length;
    for(let i = 0; i < n-1; i++){

        for(let j = 0; j < n-i-1; j++){
            
            if(array[j] > array[j+1]){
                let temp = array[j];
                array[j] = array[j+1];
                array[j+1] = temp;
            }
        }
    }
    return array;
}

let unsortedArray = [5, 3, 8, 2, 1, 9, 7];

console.log(bSort(unsortedArray)); 